accounts/src/account_service.rs

```diff
@@ -0,0 +1,302 @@
+                for _ in 0..NB_RETRY {
+                    let accounts = rpc_client
+                        .get_multiple_accounts_with_config(
+                            accounts,
+                            RpcAccountInfoConfig {
+                                encoding: Some(solana_account_decoder::UiAccountEncoding::Base64),
+                                data_slice: None,
+                                commitment: Some(CommitmentConfig::finalized()),
+                                min_context_slot: None,
+                            },
+                        )
+                        .await;
+                    match accounts {
+                        Ok(response) => {
+                            fetch_accounts = response.value;
+                            updated_slot = response.context.slot;
+                            break;
+                        }
+                        Err(e) => {
+                            // retry
+                            log::error!("Error fetching all the accounts {e:?}, retrying");
+                            continue;
+                        }
+                    }
+                }
+                for (index, account) in fetch_accounts.iter().enumerate() {
+                    if let Some(account) = account {
+                        self.account_store
+                            .initilize_account(AccountData {
+                                pubkey: accounts[index],
+                                account: account.clone(),
+                                updated_slot,
+                            })
+                            .await;
+                    }
+                }
+            }
+        }
+        log::info!("{} accounts successfully fetched", accounts.len());
+        Ok(())
+    }
+
+    pub fn process_account_stream(
+        &self,
+        mut account_stream: AccountStream,
+        mut block_stream: BlockStream,
+    ) -> Vec<AnyhowJoinHandle> {
+        let this = self.clone();
+        let processed_task = tokio::spawn(async move {
+            loop {
+                match account_stream.recv().await {
+                    Ok(account_notification) => {
+                        this.account_store
+                            .update_account(
+                                account_notification.data,
+                                account_notification.commitment,
+                            )
+                            .await;
+                    }
+                    Err(tokio::sync::broadcast::error::RecvError::Lagged(e)) => {
+                        log::error!("Account Stream Lagged by {}", e);
+                        continue;
+                    }
+                    Err(tokio::sync::broadcast::error::RecvError::Closed) => {
+                        log::error!("Account Stream Broken");
+                        break;
+                    }
+                }
+            }
+            bail!("Account Stream Broken");
```

Comment:
the sequence is a bit clunky; TBH I do not know how to do it right - maybe use return instead of break+bail:

```
log::error!("Account Stream Broken");
break;
```

```
bail!("Account Stream Broken");
```